Expo quickstart (#10134)
auth0/docs: articles/quickstart/native/react-native-expo/00-login.md

:::

<%= include('../_includes/_getting_started', { library: 'Expo'}) %>

## Install Dependencies

How to install the React Native Auth0 module.

::: note
Please refer to the [official documentation](https://docs.expo.dev/) for additional details on Expo.
:::

### Yarn

```bash
yarn add react-native-auth0
```

::: note
For further reference on yarn, check [their official documentation](https://yarnpkg.com/en/docs).
:::

### npm

```bash
npm install react-native-auth0 --save
```

The first step in adding authentication to your application is to provide a way for your users to log in. The fastest, most secure, and most feature-rich way to do this with Auth0 is to use the hosted [login page](/hosted-pages/login).

<div class="phone-mockup"><img src="/media/articles/native-platforms/ios-swift/login-ios.png" alt="Universal Login"></div>

## Integrate Auth0 in Your Application

### Setup Auth0 Config Plugin

The Auth0 package runs custom native code that needs to be configured at build time. We will use [Expo Config Plugin](https://docs.expo.dev/guides/config-plugins/) to achieve this.

> The `react-native-auth0` plugin will be added in the [Expo Config](https://docs.expo.dev/workflow/configuration/) file at `app.json` or `app.config.js`.

Reviewer: Expo just refers to configuration as "Expo config" (lowercase 'C'). Also, reworded for readability.

```suggestion
Add the `react-native-auth0` plugin to the [Expo config](https://docs.expo.dev/workflow/configuration/) file at `app.json` or `app.config.js`.
```

```json
{
  "expo": {
    ...
    "plugins": [
      [
        "react-native-auth0",
        {

>           "domain": "{DOMAIN}"

Reviewer: We should use the correct placeholder here so that the docs app is able to replace it with the domain of the tenant that the reader is currently logged in with:

```suggestion
          "domain": "${account.namespace}"
```

        }
      ]
    ]
  }
}
```

> Replace `{DOMAIN}` with the value from Auth0 application settings [as shown above](#get-your-application-keys).

Reviewer: If we use the correct placeholder above, we shouldn't need this line.

> ### Generate Native Source Code

Reviewer: Use sentence case for subheadings:

```suggestion
### Generate native source code
```

> We have to generate the native code for the above configuration to be set. To do this we will run the following command

Reviewer: Avoid "we" and prefer "you" (style guide). Also, be mindful of punctuation:

```suggestion
You must generate the native code for the above configuration to be set. To do this, run the following command:
```

```bash
expo prebuild
```

> While generating the native source code, if the [Android package](https://github.com/expo/fyi/blob/main/android-package.md) and [iOS bundle identifier](https://github.com/expo/fyi/blob/main/bundle-identifier.md) are not already present in Expo Config, you will be prompted to provide it.

Reviewer: Conciseness:

```suggestion
You will be prompted to provide the [Android package](https://github.com/expo/fyi/blob/main/android-package.md) and [iOS bundle identifier](https://github.com/expo/fyi/blob/main/bundle-identifier.md) if they are not already present in the Expo config:
```

```bash
? What would you like your Android package name to be? > com.auth0samples

? What would you like your iOS bundle identifier to be? > com.auth0samples
```

> These values are found in the Expo Config file at `app.json` or `app.config.js`. It will be used in the callback and logout URL

Reviewer: Punctuation:

```suggestion
These values are found in the Expo config file at `app.json` or `app.config.js`. It will be used in the callback and logout URLs:
```
